packages/enzyme-matchers/src/assertions/toHaveProp.js

```diff
@@ -30,11 +30,10 @@ function toHaveProp(
   ) {
     return {
       pass: props.hasOwnProperty(propKey),
-      message: `Expected <${wrapperName}> to have any value for the prop "${propKey}"`,
-      negatedMessage: `Expected <${wrapperName}> not to receive the prop "${propKey}"`,
+      message: `Expected <${wrapperName}> to have a value for prop "${propKey}", but it did not.`,
+      negatedMessage: `Expected <${wrapperName}> not to have a value for prop "${propKey}", but it did.`,
```

Comment:
I don't think this messaging is correct. These returns are for when the assertion is seeing if the component has a prop given, regardless of it's value. So it truly did not expect to `receive the prop`. Maybe the top message should be changed to `to have received the prop`

Reply:
Updated